src/io/flutter/logging/FlutterLogTree.java

```diff
@@ -706,31 +675,33 @@ public void setFilter(@NotNull FlutterLogFilterPanel.FilterParam filter) {
       return;
     }
     this.filter = filter;
-    updateRowFilter(filter);
-    updateCounter();
+    rowSorter.setRowFilter(getRowFilter(filter));
   }
 
-  private void updateRowFilter(@NotNull FlutterLogFilterPanel.FilterParam filter) {
+  private RowFilter<TableModel, Object> getRowFilter(@NotNull FlutterLogFilterPanel.FilterParam filter) {
     final String nonNullExpression = filter.getExpression() == null ? "" : filter.getExpression();
-    final String quoteRegex = filter.isRegex() ? nonNullExpression : Pattern.quote(nonNullExpression);
-
-    final String matchCaseRegex = filter.isMatchCase() ? "" : "(?i)";
-    final String standardRegex = matchCaseRegex + "(?s).*" + quoteRegex + ".*";
-    final RowFilter<TableModel, Object> logMessageFilter;
-    final RowFilter<TableModel, Object> logLevelFilter;
-    try {
-      logMessageFilter = RowFilter.regexFilter(
-        standardRegex,
-        FlutterTreeTableModel.ColumnIndex.MESSAGE.index, FlutterTreeTableModel.ColumnIndex.CATEGORY.index,
-        FlutterTreeTableModel.ColumnIndex.LOG_LEVEL.index
-      );
-    }
-    catch (PatternSyntaxException e) {
-      return;
+    if (!nonNullExpression.isEmpty() || filter.isMatchCase() || filter.isRegex()) {
+      final String quoteRegex = filter.isRegex() ? nonNullExpression : Pattern.quote(nonNullExpression);
+      final String matchCaseRegex = filter.isMatchCase() ? "" : "(?i)";
+      final String standardRegex = matchCaseRegex + "(?s).*" + quoteRegex + ".*";
+      final RowFilter<TableModel, Object> logMessageFilter;
+      final RowFilter<TableModel, Object> logLevelFilter;
+      try {
+        return RowFilter.regexFilter(
+          standardRegex,
+          FlutterTreeTableModel.ColumnIndex.MESSAGE.index,
+          FlutterTreeTableModel.ColumnIndex.CATEGORY.index,
+          FlutterTreeTableModel.ColumnIndex.LOG_LEVEL.index
+        );
+      }
+      catch (PatternSyntaxException e) {
+        // Fall through.
```

Comment:
not a new issue but we should notify the user if the regex is invalid.

Reply:
Yeah.  I'm not sure what the best UI is but ignoring isn't great.  I'll add a TODO.